picard/script/serializer.py

```diff
@@ -245,6 +250,24 @@ def create_from_yaml(cls, yaml_string, create_new_id=True):
             new_object._set_new_id()
         return new_object
 
+    @property
+    def filename(self):
+        return self.make_filename(self.title)
+
+    @classmethod
+    def make_filename(cls, title=None):
```

Comment:
Instead of a class method we could make this one a bit more generic and move it to utils:

```python
def make_filename_from_title(title=None, default=None):
    if default is None:
        default = _("No Title")
    if not title:
        title = default
    filename = sanitize_filename(title, win_compat=IS_WIN)
    if IS_WIN:
        filename = replace_win32_incompat(filename)
    return filename
```

And change caller accordingly:
```python

    @property
    def filename(self):
        return make_filename_from_title(self.title, _("Unnamed Script"))
```

It could be tested independently.

What do you think?